src/pubsub/mod_pubsub.erl

```diff
@@ -231,6 +231,9 @@ start_link(Host, Opts) ->
     Proc = gen_mod:get_module_proc(Host, ?PROCNAME),
     gen_server:start_link({local, Proc}, ?MODULE, [Host, Opts], []).
 
+deps(Host, Opts) ->
+    [{mod_caps, optional}].
```

Comment:
Isn't it the other way around? As far as I understand `mod_caps` needs `mod_pubsub` to work properly. What do I miss?

Reply:
@michalwski PEP plugin of `mod_pubsub` complains if `mod_caps` is not started beforehand: https://github.com/esl/MongooseIM/issues/1528

Reply:
That's ... convoluted. I think it would make most sense if plugins specify the modules they depends on. I understand this would be more difficult to implement, and I'm not sure it's worthy.

The least we can do to clarify the situation is to add a comment in `mod_pubsub` saying why and when it depends on `mod_caps`. Could you add such a comment @kzemek?

Reply:
[XEP-0115: Entity Capabilities](https://xmpp.org/extensions/xep-0115.html) doesn't suggest that caps depend on pubsub, only that caps can be used *by* pubsub - so I think that the optional dependency in the current direction makes sense.

Reply:
Right, this seams obvious after you read the XEP. Reading from code may not be that obvious (at least not to me) that's why I suggested to write a comment about it in the code :)